### PR TITLE
Update Sobol.jl

### PR DESCRIPTION
**Pull request by yakir12**
I'm trying to fix the bounds api for ScaledSobolSeq. Not sure if this upsets you... Check it before you accept it :)

### CODE REVIEW

**Comment by yakir12** [comment]
This works, but `next!` isn't defined for that type so there are piece missing in the pkg, which is fine. I might get to it if you'd like help.

File: src/Sobol.jl [hunk 1]
```diff
@@ -121,12 +121,14 @@ Compat.IteratorEltype(::Type{<:AbstractSobolSeq}) = Base.HasEltype()
...
+    ub::Vector{Float64}    
+    function ScaledSobolSeq(lb::Vector{Float64}, ub::Vector{Float64})
+        N = length(lb)
+        @assert N == length(ub)
         new{N}(SobolSeq(N), lb, ub)
+    end
 end
-SobolSeq(N::Integer, lb, ub) =
-    ScaledSobolSeq{Int(N)}(copy!(Vector{Float64}(undef,N), lb), copy!(Vector{Float64}(undef,N), ub))
+SobolSeq(lb, ub) = ScaledSobolSeq(Float64.(lb), Float64.(ub))
```
Comment (stevengj) [inline src/Sobol.jl]: Better to do `copy!`: broadcast is not guaranteed to produce a `Vector`.

File: src/Sobol.jl [hunk 2]
```diff
@@ -121,12 +121,14 @@ Compat.IteratorEltype(::Type{<:AbstractSobolSeq}) = Base.HasEltype()
 struct ScaledSobolSeq{N} <: AbstractSobolSeq{N}
     s::SobolSeq{N}
     lb::Vector{Float64}
-    ub::Vector{Float64}
-    ScaledSobolSeq(lb::Vector{Float64}, ub::Vector{Float64}) =
+    ub::Vector{Float64}    
+    function ScaledSobolSeq(lb::Vector{Float64}, ub::Vector{Float64})
+        N = length(lb)
+        @assert N == length(ub)
```
Comment (stevengj) [inline src/Sobol.jl]: This assertion seems weird.

Why is this change needed?

**Comment by stevengj** [comment]
What problem is being solved here?

**Comment by yakir12** [comment]
Right, so I should have started with stating what the issue is:
```julia 
julia> using Sobol

julia> s = SobolSeq(2, [0,1], [1,2])
ERROR: MethodError: no method matching copy!(::Array{Float64,1}, ::Array{Int64,1})
Stacktrace:
 [1] SobolSeq(::Int64, ::Array{Int64,1}, ::Array{Int64,1}) at /home/yakir/.julia/packages/Sobol/crr5n/src/Sobol.jl:128
 [2] top-level scope at none:0
```
I suspect I've used this incorrectly somehow.

**Comment by stevengj** [comment]
Fixed by #17.
